extensions/ql-vscode/src/extension.ts

```diff
@@ -679,7 +680,11 @@ async function activateWithInstalledDistribution(
     qs,
     distributionManager,
     databaseManager: dbm,
-    databaseUI
+    databaseUI,
+    dispose: () => {
+      ctx.subscriptions.forEach(disposable => disposable.dispose());
+    }
```

Comment:
Why do we need to explicitly write this? I expect VS Code to do something like this and dispose all `subscriptions` when the extension is deactivated. Can we make such a deactivate call during test teardown?

Reply:
Actually, we don't need this.

And, no, we cannot deactivate the extension during test teardown. We can only deactivate the extension a single time for all tests. This is _bad_ because we will be potentially sharing state across tests, but that appears to be the way it works.

This line above is left over from an attempt of mine to properly dispose after each test. But, when the disposal happened, the entire test process shut down early. So it's not possible to do proper cleanup after each test.